tests/unit-tests/plugins/woocommerce-subscriptions/class-woocommerce-subscriptions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Tests the WooCommerce Subscriptions integration class.
+ *
+ * @package Newspack\Tests
+ */
+
+use Newspack\WooCommerce_Subscriptions;
+use Newspack\Reader_Activation;
+
+/**
+ * Test WooCommerce Subscriptions integration functionality.
+ *
+ * @group WooCommerce_Subscriptions_Integration
+ */
+class Newspack_Test_WooCommerce_Subscriptions extends WP_UnitTestCase {
+	/**
+	 * Setup for the tests.
+	 */
+	public static function set_up_before_class() {
+		if ( ! defined( 'NEWSPACK_SUBSCRIPTIONS_EXPIRATION' ) ) {
+			define( 'NEWSPACK_SUBSCRIPTIONS_EXPIRATION', true );
+		}
+	}
+
+	/**
+	 * Test WooCommerce_Subscriptions::is_active.
+	 */
+	public function test_is_active() {
+		$is_active = WooCommerce_Subscriptions::is_active();
+		$this->assertTrue( $is_active, 'WooCommerce Subscriptions integration should be active when WC_Subscriptions class exists.' );
+	}
+
+	/**
+	 * Test WooCommerce_Subscriptions::is_enabled.
+	 */
+	public function test_is_enabled() {
+		$is_enabled = WooCommerce_Subscriptions::is_enabled();
+		$this->assertTrue( $is_enabled, 'WooCommerce Subscriptions integration should be enabled when Feature Flag is present.' );
```

Comment:
I wanted to test the case where FF is not defined, but could not find a way to modify globally defined constants between test runs. I tried `@preserveGlobalState` paired with `@runInSeparateProcess`, but this ended up removing other globals the tested method relied on.

Reply:
there's no good way to test constants... The best we could do would be to change the approach we use for Feature Flags and use something else to handle these things